service/src/main/java/org/mskcc/cmo/metadb/service/consumer/SampleIntakeMessageConsumer.java

```diff
@@ -1,35 +1,71 @@
 package org.mskcc.cmo.metadb.service.consumer;
 
+import com.google.gson.Gson;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.mskcc.cmo.messaging.Gateway;
 import org.mskcc.cmo.messaging.MessageConsumer;
-import org.springframework.stereotype.Component;
+import org.mskcc.cmo.metadb.persistence.SampleMetadataRepository;
+import org.mskcc.cmo.shared.neo4j.Patient;
+import org.mskcc.cmo.shared.neo4j.PatientMetadata;
+import org.mskcc.cmo.shared.neo4j.SampleMetadataEntity;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.stereotype.Service;
 
 /**
  *
  * @author ochoaa
  */
-@Component
+@Service
+@ComponentScan(basePackages = "org.mskcc.cmo.metadb.persistence")
 public class SampleIntakeMessageConsumer implements MessageConsumer {
     private Gateway messagingGateway;
     private final Log LOG = LogFactory.getLog(SampleIntakeMessageConsumer.class);
+    private SampleMetadataRepository sampleMetadataRepository;
 
-    public SampleIntakeMessageConsumer(Gateway messagingGateway) {
+    public SampleIntakeMessageConsumer(Gateway messagingGateway, 
+            SampleMetadataRepository sampleMetadataRepository) {
         this.messagingGateway = messagingGateway;
+        this.sampleMetadataRepository = sampleMetadataRepository;
     }
 
     @Override
     public void onMessage(Object message) {
+        
         LOG.info("*** Message received on igo.new-sample-intake topic ***\n");
         LOG.info(message);
+        if (addSampleMetadata(message)) {
```

Comment:
See my comments in addSampleMetadata().  I would call gson.fromJson() here and catch any exceptions before call addSampleMetadata().